lib/Recur/RRuleIterator.php

```diff
@@ -127,37 +127,54 @@ public function isInfinite()
     }
 
     /**
-     * This method allows you to quickly go to the next occurrence after the
+     * This method allows you to quickly go to the next occurrence before or after the
      * specified date.
      *
      * @param DateTimeInterface $dt
+     * @param bool $before If set to true the Iterator will be set to the occurrence before $dt
      */
-    public function fastForward(DateTimeInterface $dt)
+    public function fastForward(DateTimeInterface $dt, $before = false)
     {
+        $previousDate = null;
+
+        // We don't do any fast forwarding if we have a count limit as we cannot do jumps
         if (!isset($this->count)) {
+            $frequencyCoeff = 1;
+
+            switch ($this->frequency) {
+                case 'hourly':
+                    $frequencyCoeff = 1 / 24;
+                    break;
+                case 'daily':
+                    $frequencyCoeff = 1;
+                    break;
+                case 'weekly':
+                    $frequencyCoeff = 7;
+                    break;
+                case 'monthly':
+                    $frequencyCoeff = 30;
+                    break;
+                case 'yearly':
+                    $frequencyCoeff = 365;
+                    break;
+            }
+
+            // If we want the occurrence before, we fast forward to one frequency before the dt and then advance by one
+            // increment only to make sure that we have the previous occurrence
+            if ($before) {
+                $originalDt = clone $dt;
+                $dt = $dt->modify('- ' . ($frequencyCoeff * 24 * $this->interval) . ' hours');
+            }
+
             do {
                 $diff = $this->currentDate->diff($dt);
-                switch ($this->frequency) {
-                    case 'hourly':
-                        $i = $diff->days * 24;
-                        break;
-                    case 'daily':
-                        $i = $diff->days;
-                        break;
-                    case 'weekly':
-                        $i = $diff->days / 7;
-                        break;
-                    case 'monthly':
-                        $i = $diff->days / 30;
-                        break;
-                    case 'yearly':
-                        $i = $diff->days / 365;
-                        break;
-                }
+                $i = $diff->days / $frequencyCoeff;
+
                 $i /= $this->interval;
                 $i /= 4;
                 $i = floor($i);
                 $i = max(1, $i);
```

Comment:
`$this#next` is expecting an int. If this library is moved to php 7.1 with strict mode, a cast should be added here

Reply:
Corrected, thanks!